mbedtls: `thingy53/nrf5340/cpuapp/ns samples/net/sockets/http_server -T sample.net.sockets.https.server` error

**Describe the bug**

MbedTLS has a build error in some of the in-tree samples, leading to CI failing on unrelated changes:
See:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/14370996793/job/40294070100?pr=88397

```
/home/jordan/code/zephyr/modules/crypto/mbedtls/library/ssl_tls.c:1497:12: error: 'mbedtls_ssl_context' has no member named 'badmac_seen_or_in_hsfraglen'
 1497 |         ssl->badmac_seen_or_in_hsfraglen = 0;
```

**To Reproduce**

`west build -p -b thingy53/nrf5340/cpuapp/ns samples/net/sockets/http_server -T sample.net.sockets.https.server`

**Expected behavior**

Sample should either build successfully or be excluded from the tests if it is not expected to work.

**Impact**

CI failures

**Environment (please complete the following information):**

 - main

--- a/doc/releases/release-notes-4.2.rst
+++ b/doc/releases/release-notes-4.2.rst
@@ -555,3 +555,6 @@ Other notable changes
 
 * Updated Mbed TLS to version 3.6.3 (from 3.6.2). The release notes can be found at:
   https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.3
+
+* Updated TF-M to version 2.1.2 (from 2.1.1). The release notes can be found at:
+  https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.2/releases/2.1.2.html

--- a/submanifests/optional.yaml
+++ b/submanifests/optional.yaml
@@ -41,7 +41,7 @@ manifest:
       groups:
         - optional
     - name: tf-m-tests
-      revision: 502ea90105ee18f20c78f710e2ba2ded0fc0756e
+      revision: c712761dd5391bf3f38033643d28a736cae89a19
       path: modules/tee/tf-m/tf-m-tests
       remote: upstream
       groups:

--- a/west.yml
+++ b/west.yml
@@ -353,7 +353,7 @@ manifest:
       groups:
         - tee
     - name: trusted-firmware-m
-      revision: 857f3697ece6f44a663f007de71205a7b34974ef
+      revision: e2288c13ee0abc16163186523897e7910b03dd31
       path: modules/tee/tf-m/trusted-firmware-m
       groups:
         - tee

